tutorials/LinearAlgebra/LinearAlgebra.ipynb

```diff
@@ -0,0 +1,984 @@
+    "\n",
+    "Here is an example with relatively small matrices:\n",
+    "\n",
+    "$$\\begin{bmatrix}\n",
+    "    1 & 2 & 3 \\\\\n",
+    "    4 & 5 & 6\n",
+    "\\end{bmatrix}\n",
+    "\\begin{bmatrix}\n",
+    "    1 & 2 \\\\\n",
+    "    3 & 4 \\\\\n",
+    "    5 & 6\n",
+    "\\end{bmatrix}\n",
+    "=\n",
+    "\\begin{bmatrix}\n",
+    "    (1 \\cdot 1) + (2 \\cdot 3) + (3 \\cdot 5) & (1 \\cdot 2) + (2 \\cdot 4) + (3 \\cdot 6) \\\\\n",
+    "    (4 \\cdot 1) + (5 \\cdot 3) + (6 \\cdot 5) & (4 \\cdot 2) + (5 \\cdot 4) + (6 \\cdot 6)\n",
+    "\\end{bmatrix}\n",
+    "=\n",
+    "\\begin{bmatrix}\n",
+    "    22 & 28 \\\\\n",
+    "    49 & 64\n",
+    "\\end{bmatrix}$$\n",
+    "\n",
+    "Note that matrix multiplication is **not commutative:** $AB$ rarely equals $BA$. That being said, here are the properties it does have:\n",
+    "\n",
+    "* Associativity: $A(BC) = (AB)C$\n",
+    "* Distributivity over matrix addition: $A(B + C) = AB + AC$ and $(A + B)C = AC + BC$\n",
+    "* Associativity with scalar multiplication: $xAB = x(AB) = A(xB)$\n",
+    "\n",
+    "Another very important property of matrix multiplication is that a matrix multiplied by a vector produces another vector.\n",
+    "\n",
+    "An **identity matrix** $I_n$ is a special $n \\times n$ matrix which has $1$s on the main diagonal, and $0$s everywhere else:\n",
+    "\n",
+    "$$I_n =\n",
+    "\\begin{bmatrix}\n",
+    "    1 & 0 & 0 & \\dotsb & 0 \\\\\n",
+    "    0 & 1 & 0 & \\dotsb & 0 \\\\\n",
+    "    0 & 0 & 1 & \\dotsb & 0 \\\\\n",
+    "    \\vdots & \\vdots & \\vdots & \\ddots & \\vdots \\\\\n",
+    "    0 & 0 & 0 & \\dotsb & 1\n",
+    "\\end{bmatrix}$$\n",
+    "\n",
+    "What makes it special is that multiplying any matrix (of compatible size) by $I_n$ returns the original matrix. To put it another way, if $A$ is an $n \\times m$ matrix:\n",
+    "\n",
+    "$$AI_m = I_nA = A$$\n",
+    "\n",
+    "This is why $I_n$ is called an identity matrix - it acts as a **multiplicative identity**. In other words, it is the matrix equivalent of the number $1$."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### <span style=\"color:blue\">Exercise 3</span>: Matrix multiplication.\n",
+    "\n",
+    "**Inputs:**\n",
+    "\n",
+    "1. An $n \\times m$ matrix $A$.\n",
+    "2. An $m \\times k$ matrix $B$.\n",
+    "\n",
+    "**Goal:** Return the $n \\times k$ matrix that results from $AB$.\n",
+    "\n",
+    "<br/>\n",
+    "<details>\n",
+    "    <summary><strong>Need a hint? Click here</strong></summary>\n",
+    "    To solve this exercise, you will need 3 <code>for</code> loops: one to go over the rows in the output matrix, one to go over the columns, and one to add up the products:\n",
+    "    <pre>\n",
+    "        <code>\n",
+    "    for i in range(n):\n",
+    "        for j in range(k):\n",
+    "            sum = 0\n",
+    "            for x in range(m):\n",
```

Comment:
I'd use a different variable name for this loop to match the index we used in the formula definition of matrix multiplication:
```suggestion
    "            for t in range(m):\n",
```